Docs and link updates for plan, connect, and usage docs (#18344)
cockroachdb/docs: src/current/cockroachcloud/take-and-restore-customer-owned-backups.md

## Examples

> Before you begin, connect to your cluster. For guidance on connecting to your CockroachDB {{ site.data.products.cloud }} cluster, visit [Connect to a CockroachDB {{ site.data.products.serverless }} Cluster]({% link cockroachcloud/connect-to-a-basic-cluster.md %}) or [Connect to Your CockroachDB {{ site.data.products.dedicated }} Cluster]({% link cockroachcloud/connect-to-your-cluster.md %}).

Reviewer: ```suggestion
Before you begin, connect to your cluster. Refer to [Connect to a CockroachDB {{ site.data.products.serverless }} Cluster]({% link cockroachcloud/connect-to-a-basic-cluster.md %}) or [Connect to Your CockroachDB {{ site.data.products.dedicated }} Cluster]({% link cockroachcloud/connect-to-your-cluster.md %}).
```
Not sure about updating the variables in the link text.

Author: Yep, leaving them for now
